samples/rest/files.sh

```diff
@@ -0,0 +1,272 @@
+  -H "X-Goog-Upload-Offset: 0" \
+  -H "X-Goog-Upload-Command: upload, finalize" \
+  --data-binary "@${VIDEO_PATH}" 2> /dev/null > file_info.json
+
+file_uri=$(jq ".file.uri" file_info.json)
+echo file_uri=$file_uri
+
+curl "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=$GOOGLE_API_KEY" \
+    -H 'Content-Type: application/json' \
+    -X POST \
+    -d '{
+      "contents": [{
+        "parts":[
+          {"text": "Describe this video clip"},
+          {"file_data":{"mime_type": "video/mp4", "file_uri": '$file_uri'}}]
+        }]
+       }' 2> /dev/null > response.json
+
+cat response.json
+echo
+
+jq ".candidates[].content.parts[].text" response.json
+# [END files_create_video]
+
+echo "[START files_list]"
+# [START files_list]
+echo "My files: "
+curl "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:listFiles?key=$GOOGLE_API_KEY"
+# [END files_list]
+
+echo "[START files_get]"
+# [START files_get]
+MIME_TYPE=$(file -b --mime-type "${TEXT_PATH}")
+NUM_BYTES=$(wc -c < "${TEXT_PATH}")
+DISPLAY_NAME=TEXT
+
+tmp_header_file=upload-header.tmp
+
+# Initial resumable request defining metadata.
+# The upload url is in the response headers dump them to a file.
+curl "${BASE_URL}/upload/v1beta/files?key=${GOOGLE_API_KEY}" \
+  -D upload-header.tmp \
+  -H "X-Goog-Upload-Protocol: resumable" \
+  -H "X-Goog-Upload-Command: start" \
+  -H "X-Goog-Upload-Header-Content-Length: ${NUM_BYTES}" \
+  -H "X-Goog-Upload-Header-Content-Type: ${MIME_TYPE}" \
+  -H "Content-Type: application/json" \
+  -d "{'file': {'display_name': '${DISPLAY_NAME}'}}" 2> /dev/null
+
+upload_url=$(grep -i "x-goog-upload-url: " "${tmp_header_file}" | cut -d" " -f2 | tr -d "\r")
+rm "${tmp_header_file}"
+
+curl "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:getFile?key=$GOOGLE_API_KEY" /
+    -H 'Content-Type: application/json' \
+    -X POST \
+    -d '{
+      "name": '${DISPLAY_NAME}'}}"
+       }' 2> /dev/null > response.json
+# [END files_get]
+
+echo "[START files_delete]"
+# [START files_delete]
+MIME_TYPE=$(file -b --mime-type "${TEXT_PATH}")
+NUM_BYTES=$(wc -c < "${TEXT_PATH}")
+DISPLAY_NAME=TEXT
+
+tmp_header_file=upload-header.tmp
+
+# Initial resumable request defining metadata.
+# The upload url is in the response headers dump them to a file.
+curl "${BASE_URL}/upload/v1beta/files?key=${GOOGLE_API_KEY}" \
+  -D upload-header.tmp \
+  -H "X-Goog-Upload-Protocol: resumable" \
+  -H "X-Goog-Upload-Command: start" \
+  -H "X-Goog-Upload-Header-Content-Length: ${NUM_BYTES}" \
+  -H "X-Goog-Upload-Header-Content-Type: ${MIME_TYPE}" \
+  -H "Content-Type: application/json" \
+  -d "{'file': {'display_name': '${DISPLAY_NAME}'}}" 2> /dev/null
+
+upload_url=$(grep -i "x-goog-upload-url: " "${tmp_header_file}" | cut -d" " -f2 | tr -d "\r")
+rm "${tmp_header_file}"
+
+curl "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:deleteFile?key=$GOOGLE_API_KEY" /
```

Comment:
Same, what if we add "delete_file" at the end of "files_create_text" so we don't need to create it again for the example?